### ISSUE TITLE
cat can't expand the number of dimensions but doesn't produce an error

### ISSUE THREAD

**Issue by gustafsson** [JuliaGPU/CuArrays.jl#73]
Maybe related to #72 but the behaviour is different. There is no error, but the result of `cat(3,cu(x),cu(y))` is different from `cat(3,x,y)`:

```
julia> using CuArrays;

julia> x = rand(2, 1);

julia> y = rand(2, 1);

julia> cat(3,x,y)
2×1×2 Array{Float64,3}:
[:, :, 1] =
 0.151418
 0.388829

[:, :, 2] =
 0.732151
 0.991128

julia> cat(3,cu(x),cu(y))
2×1×2 CuArray{Float32,3}:
[:, :, 1] =
 0.151418
 0.388829

[:, :, 2] =
 -1.25053f-6
  3.89817f-6
```

**Comment by gustafsson** [comment]
I think this error stems from [this line](https://github.com/JuliaGPU/CuArrays.jl/blob/v0.5.0/src/utils.jl#L84) where `getindex.(shapes, dim)` assumes that `dim` is a valid index in all `shapes`. In my example above `length.(shapes)==[2,2]` but `dim==3`. This causes a buffer over-read which produces the garbage numbers (`-1.25053f-6, 3.89817f-6`) seen in the result above.

**Comment by gustafsson** [comment]
Closed by #75
